File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

float sentences;

//functions
int count_letters(string words);
int count_words(string words);
int count_sentences(string words);

int main(void)
{
    //getting string
    string text = get_string("Text: ");

    //length of string
    float letters = count_letters(text);

    //sentences in string
    sentences = count_sentences(text);

    //words in string
    float words = count_words(text);


    //calculating index
    float l = (letters / words) * 100;
    float s = (sentences / words) * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}




//counting letters
int count_letters(string words)
{
    int length = strlen(words);
    int count = 0;
    for (int i = 0; i < length; i++)
    {
        int temp = toupper(words[i]);
        if (temp >= 65 && temp <= 90)
        {
            count++;
        }
        else
        {
            continue;
        }
    }
    return count;
}

//counting words
int count_words(string words)
{
    int length = strlen(words);
    int count = 0;
    for (int i = 0; i < length; i++)
    {
        int temp = words[i];
        //looking for full stops spaces question marks or exclamation marks
        if (temp == 32 || temp == 33 || temp == 46 || temp == 63)
        {
            count++;
        }
        else
        {
            continue;
        }
    }
    count = count - sentences + 1;

    return count;
}

//counting sentences
int count_sentences(string words)
{
    int length = strlen(words);
    int count = 0;
    for (int i = 0; i < length; i++)
    {
        int temp = toupper(words[i]);
        //looking for questions marks periods and exclamation marks
        if (temp == 33 || temp == 46 || temp == 63)
        {
            count++;
        }
        else
        {
            continue;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
import curses.ascii

# count number of letters


def countletters(phrase):
    count = 0
    for i in range(len(phrase)):
        if curses.ascii.isalpha(phrase[i]):
            count += 1
        else:
            continue
    return count


# count number of words


def countwords(phrase):
    temp = phrase.split()
    return len(temp)

# count number of sentences


def countsentences(phrase):
    count = 0
    for i in range(len(phrase)):
        if phrase[i] in (".", "?", "!"):
            count += 1
        else:
            continue
    return count


# get input
text = input("Text: ")

# variables to use in formula
letternum = countletters(text)
wordsnum = countwords(text)
sentencesnum = countsentences(text)

L = (letternum/wordsnum) * 100
S = (sentencesnum/wordsnum) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# print result
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

print(letternum)
print(wordsnum)
print(sentencesnum)